src/test/scala/com/databricks/spark/csv/CsvSuite.scala

```diff
@@ -419,14 +416,30 @@ class CsvSuite extends FunSuite {
       .collect()
 
     val expected =
-      Seq(Seq("1", "2", "3", "4", "5"),
-          Seq("6", "7", "8", "9", "0"),
-          Seq("1", "2", "3", "4", "5"))
+      Seq(Seq("1", "2", "3", "4", "5.01", "2015-08-20 15:57:00"),
+          Seq("6", "7", "8", "9", "0", "2015-08-21 16:58:01"),
+          Seq("1", "2", "3", "4", "5", "2015-08-23 18:00:42"))
+
+    assert(results.toSeq.map(_.toSeq) === expected)
+  }
+
+  test("Inferring schema") {
+    val results: Array[Row] = new CsvParser()
+      .withDelimiter(',')
+      .withComment('~')
+      .withInferSchema(true)
+      .csvFile(TestSQLContext, commentsFile)
+      .collect()
+
+    val expected =
```

Comment:
what's the difference between this suite and CsvFastSuite? Not necessary in this PR, but maybe you can define an abstract test suite and have CsvFastSuite and CsvSuite extend it with different configurations?